src/v1/app.files.ts

```diff
@@ -247,7 +247,12 @@ export function initFileTree() {
         setEnabled(uiFiles.cloneKsyFile, isLocal && isKsy);
         setEnabled(uiFiles.deleteItem, isLocal);
         setEnabled(uiFiles.generateParser, isKsy);
-        uiFiles.fileTreeContextMenu.css({ display: "block", left: e.pageX, top: e.pageY });
+
+        uiFiles.fileTreeContextMenu.css({ display: "block", visibility: "hidden", "z-index": -1 }); // still invisible but has width & height
+        var x = Math.min(e.pageX, $(window).width() - uiFiles.fileTreeContextMenu.width());
+        var h = uiFiles.fileTreeContextMenu.height();
+        var y = e.pageY > ($(window).height() - h) ? e.pageY - h : e.pageY;
+        uiFiles.fileTreeContextMenu.css({ visibility: '', "z-index": '', left: x, top: y });
```

Comment:
I think setting explicit values would be clearer, default values are `visibility: visible` and `z-index: auto`.

Reply:
This unfortunately doesn't work, this prevents the context menu from appearing. There won't be any problem with `visibility: visible` currently, because the external CSS styles are not using it, but [they set `z-index` to `9999`](https://github.com/kaitai-io/kaitai_struct_webide/blob/2a967590d413e6995e04f096b7f2168cca374c6e/css/app.css#L26) (which is needed to overlay everything on the page coming later in the HTML). So I can't use `z-index: auto`, because it will be covered:

![contextmenu-z-index-auto](https://user-images.githubusercontent.com/47499687/76689946-86451780-663b-11ea-84bd-dbd5bf933cfa.gif)

So I consider unsetting both `visibility` and `z-index` (that's done by setting them to the empty string `''` according to [the jQuery documentation](https://api.jquery.com/css/)) as the safest way how to restore things to their original state, because it works even if the external styles are relying on them.

Reply:
I didn't know that unsetting was done with an empty string. Makes sense when thinking about it from a HTML inline style perspective though, which of course is what `element.style.whatever` would refer to.